tests/unit/s2n_server_extensions_test.c

```diff
@@ -361,7 +382,13 @@ int main(int argc, char **argv)
             EXPECT_SUCCESS(s2n_enable_tls13());
             struct s2n_connection *conn;
             EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_SERVER));
-            EXPECT_SUCCESS(s2n_connection_allow_all_response_extensions(conn));
+            s2n_extension_type_id extension_id = 0;
+            EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_KEY_SHARE, &extension_id));
+            S2N_CBIT_SET(conn->extension_requests_received, extension_id);
+            EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_SUPPORTED_VERSIONS, &extension_id));
+            S2N_CBIT_SET(conn->extension_requests_received, extension_id);
+            EXPECT_SUCCESS(s2n_extension_supported_iana_value_to_id(TLS_EXTENSION_SESSION_TICKET, &extension_id));
+            S2N_CBIT_SET(conn->extension_requests_received, extension_id);
```

Comment:
If you're going to change this, why not only turn on the session ticket extension and update the expected size? That seems more in line with what the test wants to do. Same for the other tests that are trying to test a single extension.